kernel/mount.c

```diff
@@ -299,6 +310,114 @@ static const char* _find_arg(const char* args[], const char* name)
 }
 #endif /* MYST_ENABLE_EXT2FS */
 
+#ifdef MYST_ENABLE_HOSTFS
+static bool _find_mount_source(const char* source, char path[PATH_MAX])
+{
+    bool found = false;
+    for (size_t i = 0; i < _mount_table_size; i++)
+    {
+        if (strcmp(_mount_table[i].source, source) == 0)
+        {
+            found = true;
+            myst_strlcpy(path, _mount_table[i].path, PATH_MAX);
+            break;
+        }
+    }
+    return found;
+}
+
+static long _new_automount_dir(char mountdir[PATH_MAX])
+{
+    long ret = -1;
+    int mount_num = 1;
+
+    myst_strlcpy(mountdir, AUTOMOUNT_DIR, PATH_MAX);
+    if (!_created_automount_dir)
+    {
+        ECHECK(myst_syscall_mkdir(mountdir, 0x777));
+        _created_automount_dir = true;
+    }
+
+    while (1)
+    {
+        snprintf(mountdir, PATH_MAX, "%s/%d", AUTOMOUNT_DIR, mount_num++);
+        if (myst_syscall_access(mountdir, F_OK) != 0)
```

Comment:
This loop seems only to terminate if the mountdir does not exist.

Reply:
Yes, it is what is intended -- the code jumps out of the loop once it finds a non-existence mount folder, e.g. ${AUTOMOUNT_DIR}/1, ${AUTOMOUNT_DIR}/2, ...

Reply:
I'm sorry. I missed that. :)